openmls/src/framing/mls_auth_content_in.rs

```diff
@@ -158,6 +155,26 @@ impl AuthenticatedContentIn {
     }
 }
 
+/// Note: we can't `derive(tls_codec::Deserialize)` because [`FramedContentAuthData`] cannot
+///       implement the usual `tls_codec::Deserialize` as it requires the content type as parameter.
+impl tls_codec::Deserialize for AuthenticatedContentIn {
```

Comment:
Previously this was `AuthenticatedContent`. Now, we use `AuthenticatedContentIn` and a test-gated conversion to `AuthenticatedContent`.